core/node/bitswap.go

```diff
@@ -71,12 +74,24 @@ type bitswapIn struct {
 // Additional options to bitswap.New can be provided via the "bitswap-options"
 // group.
 func Bitswap(provide bool) interface{} {
-	return func(in bitswapIn, lc fx.Lifecycle) *bitswap.Bitswap {
+	return func(in bitswapIn, lc fx.Lifecycle) (*bitswap.Bitswap, error) {
 		bitswapNetwork := bsnet.NewFromIpfsHost(in.Host)
 
 		var provider routing.ContentDiscovery
 		if provide {
-			provider = in.Rt
+			// We need to hardcode the default because it is an
+			// internal setting in boxo.
+			pqm, err := rpqm.New(bitswapNetwork,
+				in.Rt,
+				rpqm.WithMaxProviders(10),
```

Comment:
@hsanjuan  why do we hardcode 10 here?

Wasn't previous default `DefaultMaxProviders = 0` from `boxo/routing/providerquerymanager/providerquerymanager.go`?

Reply:
Yes, this is the value we always used:
https://github.com/ipfs/boxo/blob/main/bitswap/client/client.go#L209

When passing in the pqm explicitally we need to set it...

No change on that front. As to why 10... :shrug: In rainbow we do unlimited, but in Kubo we never ventured to change the default of 10.

Reply:
Thanks! Feels we should avoid hardcoding magic numbers without docs, opened  https://github.com/ipfs/kubo/pull/10773 to sort this out + maybe align with rainbow.